yoti-sdk-impl/src/main/java/com/yoti/api/client/docs/DocScanService.java

```diff
@@ -0,0 +1,260 @@
+    private final String apiUrl;
+
+    private DocScanService(UnsignedPathFactory pathFactory, ObjectMapper objectMapper) {
+        this.unsignedPathFactory = pathFactory;
+        this.objectMapper = objectMapper;
+
+        apiUrl = System.getProperty(PROPERTY_YOTI_DOCS_URL, DEFAULT_YOTI_DOCS_URL);
+    }
+
+    public static DocScanService newInstance() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        return new DocScanService(new UnsignedPathFactory(), objectMapper);
+    }
+
+    /**
+     * Uses the supplied session specification to create a session
+     *
+     * @param appId       the app ID
+     * @param keyPair     the {@code KeyPair}
+     * @param sessionSpec the {@code SessionSpec}
+     * @return the session creation result
+     * @throws DocScanException if there was an error
+     */
+    public CreateSessionResult createSession(String appId, KeyPair keyPair, SessionSpec sessionSpec) throws DocScanException {
+        notNullOrEmpty(appId, "Application id");
+        notNull(keyPair, "Application key Pair");
+        notNull(sessionSpec, "sessionSpec");
+
+        String path = unsignedPathFactory.createNewYotiDocsSessionPath(appId);
+        LOG.info("Creating session at '{}'", path);
+
+        try {
+            byte[] payload = objectMapper.writeValueAsBytes(sessionSpec);
+
+            SignedRequest signedRequest = getSignedRequestBuilder()
+                    .withKeyPair(keyPair)
+                    .withBaseUrl(apiUrl)
+                    .withEndpoint(path)
+                    .withHttpMethod(HTTP_POST)
+                    .withPayload(payload)
+                    .withHeader(CONTENT_TYPE, CONTENT_TYPE_JSON)
+                    .build();
+
+            return signedRequest.execute(SimpleCreateSessionResult.class);
+        } catch (GeneralSecurityException ex) {
+            throw new DocScanException("Error signing the request: " + ex.getMessage(), ex);
+        } catch (ResourceException ex) {
+            throw new DocScanException("Error posting the request: " + ex.getMessage(), ex);
+        } catch (IOException | URISyntaxException ex) {
+            throw new DocScanException("Error building the request: " + ex.getMessage(), ex);
+        } catch (Exception ex) {
+            throw new DocScanException("Error creating the session: " + ex.getMessage(), ex);
+        }
+    }
+
+    /**
+     * Retrieves the current state of a given session
+     *
+     * @param appId     the app ID
+     * @param keyPair   the {@code KeyPair}
+     * @param sessionId the session ID
+     * @return the session state
+     * @throws DocScanException if there was an error
+     */
+    public SimpleGetSessionResult retrieveSession(String appId, KeyPair keyPair, String sessionId) throws DocScanException {
```

Comment:
Is there a reason this is `retrieveSession()`, rather than `getSession()`?
_(do other SDKs also need to be `retrieveSession()`)_

Reply:
I don't think we have to make sure they're exactly the same, as long as the name represents the same task (i.e. retrieve and get are pretty much the same thing)